src/community/script/core/src/main/java/org/geoserver/script/rest/service/ScriptService.java

```diff
@@ -0,0 +1,214 @@
+/** (c) 2014 Open Source Geospatial Foundation - all rights reserved
+ * (c) 2001 - 2013 OpenPlans
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ *
+ * @author David Vick, Boundless 2017
+ **/
+package org.geoserver.script.rest.service;
+
+import com.google.common.collect.Lists;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.geoserver.ows.util.ResponseUtils;
+import org.geoserver.platform.resource.Resource;
+import org.geoserver.platform.resource.Resource.Type;
+import org.geoserver.platform.resource.Resources;
+import org.geoserver.rest.RestException;
+import org.geoserver.rest.util.IOUtils;
+import org.geoserver.script.ScriptManager;
+import org.geoserver.script.rest.model.Script;
+import org.geoserver.util.Filter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.util.List;
+
+import static java.lang.String.format;
+
+@Service
+public class ScriptService {
+
+    @Autowired
+    ScriptManager scriptMgr;
+
+    String path;
+
+    public List<Script> getScriptList(HttpServletRequest request) {
+        final String type = (String) request.getAttribute("type");
+        this.path = getPath(request);
+        this.path = stripExtension(this.path);
+
+        org.geoserver.platform.resource.Resource dir = scriptMgr.script(path);
```

Comment:
Don't think you need the full package name anymore?

Reply:
fixed.